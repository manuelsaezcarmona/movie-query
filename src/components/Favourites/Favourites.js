import MovieCard from '../MovieCard/MovieCard';

export default function Favourites() {
  return (
    <main className="main-container">
      <h2>Mis Peliculas Favoritas</h2>
      <section className="favourites">
        <p className="favourites__text">No se han encontrado Peliculas</p>
        <ul className="favourites__list">
          <li className="favourites__item">
            <MovieCard />
          </li>
        </ul>
      </section>
    </main>
  );
}
