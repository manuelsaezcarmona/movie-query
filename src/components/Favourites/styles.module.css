.header {
  margin: 0.5em 0;
  padding: 0.5em 0;
  text-align: center;
  box-shadow: 0px 6px 10px 0px rgba(0, 0, 0, 0.1);
}
.favourites__list {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(315px, 1fr));
  gap: 6px;
  list-style-type: none;
  margin: 0;
  padding: 0;
}
.favourites__text {
  font-family: var(--navbar-font);
  font-size: 2em;
  text-align: center;
  margin: 20vh 0;
}

@media (min-width: 769px) {
  .favourites__text {
    font-size: 3em;
  }
}
