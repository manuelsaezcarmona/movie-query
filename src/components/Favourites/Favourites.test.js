import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';
import Favourites from './Favourites';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
  user: {
    username: 'test',
    favourites: []
  },
  movie: {
    page: 1,
    movies: [],
    activemovie: {},
    totalpages: 3,
    detailedmovie: {}
  },
  settings: {
    currentpage: 1,
    searchfilm: 'Avengers'
  }
};
const store = mockStore(initialState);
store.dispatch = jest.fn();

describe('given the Favourites Component', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <Favourites />
      </Provider>
    );
  });
  test('The component is rendered (take a snapshot)', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Favourites />
      </Provider>
    );
    expect(asFragment(<Favourites />)).toMatchSnapshot();
  });
});
