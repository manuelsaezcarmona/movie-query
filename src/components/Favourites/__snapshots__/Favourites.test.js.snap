// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given the Favourites Component The component is rendered (take a snapshot) 1`] = `
<DocumentFragment>
  <main
    class="main-container"
  >
    <h2
      class="header"
    >
      Mis Peliculas Favoritas
    </h2>
    <section
      class="favourites"
    >
      <p
        class="favourites__text"
      >
        No tienes peliculas Favoritas
      </p>
    </section>
  </main>
</DocumentFragment>
`;
